src/Compilers/VisualBasic/Portable/Lowering/Rewriter.vb

```diff
@@ -73,14 +75,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                                             lazyVariableSlotAllocator,
                                                             compilationState,
                                                             If(symbolsCapturedWithoutCopyCtor, SpecializedCollections.EmptySet(Of Symbol)),
-                                                            diagnostics,
+                                                            localDiagnostics,
                                                             rewrittenNodes)
             End If
 
-            If bodyWithoutLambdas.HasErrors Then
+            If bodyWithoutLambdas.HasErrors OrElse localDiagnostics.HasAnyErrors Then
+                diagnostics.AddRangeAndFree(localDiagnostics)
```

Comment:
This statement frees localDiagnostics, so we won't be able access it in the ```If``` condition.